### ISSUE TITLE
ProtocolError undefined

### ISSUE THREAD

**Issue by mattdesl** [livereload/livereload-js#35]
I'm getting a fair bit of errors in Chrome where ProtocolError is undefined. The injected LiveReload script is v2.2.2.

Problematic line in Connector:

``` js
        error: (function(_this) {
          return function(e) {
            if (e instanceof ProtocolError) {
              if (typeof console !== "undefined" && console !== null) {
                return console.log("" + e.message + ".");
              }
            } else {
              if (typeof console !== "undefined" && console !== null) {
                return console.log("LiveReload internal error: " + e.message);
              }
            }
          };
        })(this)
```

**Comment by rickihastings** [comment]
I'm getting this too, any update on this?

**Comment by kursion** [comment]
Still nothing ? :(

**Comment by call-a3** [comment]
+1

**Comment by webfella** [comment]
+ 1

**Comment by patuwwy** [comment]
```
e
ProtocolError {
    message:"LiveReload protocol error (invalid command 'hello', only valid commands are: reload, alert)) after receiving data: "{"command":"hello","protocols":["http://livereload.com/protocols/official-7"],"serverName":"atom-livereload"}"."
}
```

﻿

**Comment by charleshimmer** [comment]
Yes I get this from time to time.  A refresh normally makes it go away but it would be nice to know why it's happening or find a fix.

**Comment by spenat28** [comment]
Can it be connected to asynchronous load of livereload.js? I didn't have this issue untill we started using async.

**Comment by mprobst** [comment]
I think the problem is simply that this file here doesn't import `ProtocolError`:
https://github.com/livereload/livereload-js/blob/03b09762e930f6c6c67ee270c208d1c11de0247f/src/livereload.coffee#L56

**Comment by Arrow7000** [comment]
I am getting this error repeatedly. Almost every single refresh causes loads of these errors to appear. Is any progress being made on this?

Thanks

**Comment by jbhatt091** [comment]
@charleshimmer you saved my time. I got this error for the first time and simple refresh worked for me. Thanks :)

**Comment by smhg** [comment]
(I just joined the project in an effort to get things going again)

The fix is in master and will soon be released as version 2.2.3.

**Comment by smhg** [comment]
@patuwwy The error you mentioned was causing this bug to be triggered. But it is a separate issue, which is likely caused by the livereload **server** you use.
